crates/wallet/src/wallet/tx_details.rs

```diff
@@ -0,0 +1,83 @@
+//! `TxDetails`
+
+use alloc::vec::Vec;
+use serde::{Deserialize, Serialize};
+
+use bitcoin::Txid;
+
+use crate::collections::HashMap;
+
+/// Kind of change that can be applied to [`TxDetails`]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Record {
+    /// tx canceled
+    Canceled,
+    /// new details added. note, when applied this will overwrite existing
+    Details(Details),
+}
```

Comment:
I don't really understand the design of this enum.

What is the difference between `Record::Canceled` and `Record::Details { canceled: true}`.
Why are both structs required?